### PR TITLE
step2code

### PR DESCRIPTION
**Pull request by ShivanshSinghnnovative**

### CODE REVIEW

File: src/App.vue [hunk 1]
```diff
@@ -1,52 +1,87 @@
 <template>
-    <div class="header">
-        Search Hashtag :
-        <input type="text">
-    </div>
+    <searchHastag @searchHashtag="updateHashTagFilter">
+        <template v-slot:search>
+            Search Hashtag :
+            <input type="text" v-model="searchBar">
```
Comment (PiyushInnotech) [inline src/App.vue]: use searchterm instead

File: src/App.vue [hunk 2]
```diff
@@ -1,52 +1,87 @@
...
+
 
 const blogData = ref(blogsDatas);
+const searchBar = ref('');
+
+const updateSearchHashtag = (topic) => {
+    searchBar.value = topic;
+};
+
+const updateHashTagFilter = (hashtag) => {
```
Comment (PiyushInnotech) [inline src/App.vue]: use one function

File: src/App.vue [hunk 3]
```diff
@@ -1,52 +1,87 @@
...
 import cardDetails from './components/cardDetails.vue'
+import searchHastag from './components/searchHastag.vue'
+import hastagLists from './components/hastagLists.vue'
 import blogsDatas from './assets/blogData.json'
-import { ref } from 'vue'
+import { ref, computed } from 'vue'
+
 
 const blogData = ref(blogsDatas);
+const searchBar = ref('');
```
Comment (PiyushInnotech) [inline src/App.vue]: change the name

File: src/components/searchHastag.vue [hunk 4]
```diff
@@ -0,0 +1,19 @@
+<template>
+  <div class="header">
+    <slot name="search"></slot>
+  </div>
+</template>
+
+<script setup>
+import { ref, watch, defineEmits } from 'vue';
+
+const searchBar = ref('');
```
Comment (PiyushInnotech) [inline src/components/searchHastag.vue]: use props

File: src/components/hastagLists.vue [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<template>
+    <div class="topic" v-for="topic in blog.topics" :key="topic.id" @click="clickedTopic(topic)">
+      <a>  # {{ topic }}</a>
+    </div>
+</template>
+  
+<script setup>
+import { defineProps, defineEmits } from 'vue';
+const props = defineProps({
+    blog: Object,
```
Comment (meanstack9716) [inline src/components/hastagLists.vue]: hashtag component should only have topics/hashtags not the full data @ShivanshSinghnnovative

File: src/components/hastagLists.vue [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+});
+console.log(props);
```
Comment (meanstack9716) [inline src/components/hastagLists.vue]: remove console @ShivanshSinghnnovative

File: src/components/hastagLists.vue [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+
+const emits = defineEmits(['clickedTopic']);
+
+const clickedTopic = (topic) => {
+    emits('clickedTopic', topic);
+};
+</script>
+
+<style>
+a{
```
Comment (meanstack9716) [inline src/components/hastagLists.vue]: code format issue @ShivanshSinghnnovative

File: src/components/searchHastag.vue [hunk 8]
```diff
@@ -0,0 +1,23 @@
+<template>
+  <div class="header">
+    <slot name="search"></slot>
```
Comment (meanstack9716) [inline src/components/searchHastag.vue]: every time you do not need to create a slot component. 
slot components are used only when the HTML elements are dynamic. In searchHastag component slot will not be dynamic @ShivanshSinghnnovative 
This component will be a search component which can be used to show an input filed and emit a function 
 with search teaxt whenever something write in input. @ShivanshSinghnnovative

**Review by meanstack9716 — CHANGES_REQUESTED** [review]

**Pull request merged**